src/lib/decks.ts

```diff
@@ -6,7 +6,20 @@ import hospitalJson from '$lib/crc-decks/hospital.json';
 
 export const withId: <T extends object>(o: T) => Keyed<T> = (function () {
   let nextId = 0;
-  return (o) => ({ id: nextId++, ...o });
+  let skipIds = new Set();
+  return (o) => {
+    if ("id" in o) {
+      let id = Number(o.id);
+      if (id != 0 && !id) {
+        throw TypeError(`Encountered non-numeric id!: ${o.id} in ${o}`);
+      }
+      skipIds.add(id);
+    }
+    while (skipIds.has(nextId)) {
+      skipIds.delete(nextId++);
+    }
+    return { id: nextId++, ...o };
+  };
```

Comment:
A couple thoughts:
- do we know what happens on `Number(o.id)` if it's not a number?
- is there some (more wasteful but faster) algorithm for determining valid ids, like `Math.max` in the first sol
- an intended function of withId is that it would clobber an existing id field with a valid unique id. idk if this behavior was actually ever depended on... may be worth a check.

Reply:
1. Basic testing confirms classic JS ingenuity: `Number(null) == 0`, `Number("some string") == NaN`, `Number(undefined) == NaN`
2. This seems obvious in hindsight! good catch
3. It doesn't seem like this functionality was used: 
![image](https://github.com/user-attachments/assets/b5e971f2-19b6-4b18-aff6-6e2f5d1e49ed)

That said, I'm not sure how to determine if an id is invalid (and not just old) without keeping a reference to every object tagged with an ID.